src/contracts/misc/GhoGsmSteward.sol

```diff
@@ -0,0 +1,137 @@
+
+  mapping(address => GsmDebounce) internal _gsmTimelocksByAddress;
+
+  /**
+   * @dev Only methods that are not timelocked can be called if marked by this modifier.
+   */
+  modifier notTimelocked(uint40 timelock) {
+    require(block.timestamp - timelock > MINIMUM_DELAY, 'DEBOUNCE_NOT_RESPECTED');
+    _;
+  }
+
+  /**
+   * @dev Constructor
+   * @param gsmFeeStrategyFactory The address of the GSM Fee Strategy Factory
+   * @param riskCouncil The address of the risk council
+   */
+  constructor(
+    address gsmFeeStrategyFactory,
+    address riskCouncil
+  ) RiskCouncilControlled(riskCouncil) {
+    require(gsmFeeStrategyFactory != address(0), 'INVALID_GSM_FEE_STRATEGY_FACTORY');
+
+    GSM_FEE_STRATEGY_FACTORY = gsmFeeStrategyFactory;
+  }
+
+  /**
+   * @inheritdoc IGhoGsmSteward
+   */
+  function updateGsmExposureCap(
+    address gsm,
+    uint128 newExposureCap
+  ) external onlyRiskCouncil notTimelocked(_gsmTimelocksByAddress[gsm].gsmExposureCapLastUpdated) {
+    uint128 currentExposureCap = IGsm(gsm).getExposureCap();
+    require(
+      _isDifferenceLowerThanMax(currentExposureCap, newExposureCap, currentExposureCap),
+      'INVALID_EXPOSURE_CAP_UPDATE'
+    );
+
+    _gsmTimelocksByAddress[gsm].gsmExposureCapLastUpdated = uint40(block.timestamp);
+
+    IGsm(gsm).updateExposureCap(newExposureCap);
+  }
+
+  /**
+   * @inheritdoc IGhoGsmSteward
+   */
+  function updateGsmBuySellFees(
+    address gsm,
+    uint256 buyFee,
+    uint256 sellFee
+  ) external onlyRiskCouncil notTimelocked(_gsmTimelocksByAddress[gsm].gsmFeeStrategyLastUpdated) {
+    address currentFeeStrategy = IGsm(gsm).getFeeStrategy();
+    require(currentFeeStrategy != address(0), 'GSM_FEE_STRATEGY_NOT_FOUND');
+
+    uint256 currentBuyFee = IGsmFeeStrategy(currentFeeStrategy).getBuyFee(1e4);
+    uint256 currentSellFee = IGsmFeeStrategy(currentFeeStrategy).getSellFee(1e4);
+    require(
+      _isDifferenceLowerThanMax(currentBuyFee, buyFee, GSM_FEE_RATE_CHANGE_MAX),
+      'INVALID_BUY_FEE_UPDATE'
+    );
+    require(
+      _isDifferenceLowerThanMax(currentSellFee, sellFee, GSM_FEE_RATE_CHANGE_MAX),
+      'INVALID_SELL_FEE_UPDATE'
+    );
+
+    IGsmFeeStrategyFactory strategyFactory = IGsmFeeStrategyFactory(GSM_FEE_STRATEGY_FACTORY);
+    uint256[] memory buyFeeList = new uint256[](1);
+    uint256[] memory sellFeeList = new uint256[](1);
+    buyFeeList[0] = buyFee;
+    sellFeeList[0] = sellFee;
+    address strategy = strategyFactory.createStrategies(buyFeeList, sellFeeList)[0];
+
+    _gsmTimelocksByAddress[gsm].gsmFeeStrategyLastUpdated = uint40(block.timestamp);
+
+    IGsm(gsm).updateFeeStrategy(strategy);
+  }
+
+  /**
+   * @inheritdoc IGhoGsmSteward
+   */
+  function getGsmTimelocks(address gsm) external view returns (GsmDebounce memory) {
+    return _gsmTimelocksByAddress[gsm];
+  }
+
+  /**
+   * @inheritdoc IGhoGsmSteward
+   */
+  function getGsmFeeStrategies() external view returns (address[] memory) {
```

Comment:
we probably do not need this anymore, because can be fetched via factory already

Reply:
Done